### DISCUSSION TITLE
Red X in the status line: ❌Saved

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#744] [Help]
I'm getting a red X in the status line, on autosaves, manual saves, or when hovering over images (and the link is shown). Eg:
![image](https://user-images.githubusercontent.com/77707706/195645300-eef5a709-97da-4095-8b3d-bcd0a8890a7e.png)

The file seems to be saved normally, though. What does the red X means?

**Comment by euu2021** [comment]
After restarting FP, things get back to normal. But, now, I'm getting an exclamation sign:
![image](https://user-images.githubusercontent.com/77707706/197346870-5b4db852-dc5e-4deb-8fa3-9ce7aed86ba5.png)

I guess it's a bug, right? I will try to understand when it happens, and create an issue.

**Reply by dpolivaev** [reply]
It should not be like that.

**Comment by euu2021** [comment]
Ok, I finally found the source of this behavior: the Bookmarks add-on. Once the Bookmarks add-on is used in the map, those icons persistently appear in the status line. I will report at the add-on repository.

**Comment by lilive** [comment]
Hi,
The problem is:
When a script display a message in the status bar, with an icon, for example
```
c.setStatusInfo( 'standard', 'A message with a warning icon', 'messagebox_warning' )
```
then this icon is shown next to any subsequent "Saved" message.

The attached map contains all you need to quickly reproduce this behavior.
[Status bar icon problem.zip](https://github.com/freeplane/freeplane/files/10999916/Status.bar.icon.problem.zip)

**Comment by lilive** [comment]
I read the API documentation:

> void setStatusInfo(String infoPanelKey, String info)
> Info for status line, null to remove. **Removes icon if there is one**.

So, maybe this behavior is deliberate.
If so, I'm wondering how scripts are supposed to display messages with icons. Perhaps they should not use the 'standard' infoPanelKey, but use their own. Something like
```
c.setStatusInfo( 'bookmarks_addon', 'A message with a warning icon', 'messagebox_warning' )
```
in this case.

I just tried. It works. The problem is that the message stay forever, and that we need a
```
c.setStatusInfo( 'bookmarks_addon', '' )
```
to delete it.

Do I understand right ?

**Reply by EdoFro** [reply]
yes. 

you can use Timer runAfter() to delete it after a few seconds

look at this:
https://blog.mrhaki.com/2009/11/groovy-goodness-run-code-at-specified.html?m=1

or search for groovy runAfter in internet.

**Reply by EdoFro** [reply]
here is an example of how I used it:

https://github.com/EdoFro/Freeplane_Map-Drive-Inator/blob/4a400c9b6b850e961f166818629bccfd646db8e7/MapDriveInator/src/main/groovy/MDI.groovy#L785

**Reply by lilive** [reply]
Thank you Edo !

**Reply by lilive** [reply]
I have wrote a slightly modified version of your function to avoid unwanted deletion when messages are too close.

```groovy
import org.freeplane.plugin.script.proxy.ScriptUtils
import java.util.TimerTask

public class Utils
{
    private static Timer timer = new Timer()
    private static TimerTask timerTask = null
    
    public static void setStatusInfo( String message, String icon = '' )
    {
        if( timerTask ) timerTask.cancel()

        if( icon ){
            ScriptUtils.c().setStatusInfo( 'lilive bookmarks',  message , icon )
        } else {
            ScriptUtils.c().setStatusInfo( 'lilive bookmarks',  message )
        }
        timerTask = timer.runAfter(
            5000,
            { ScriptUtils.c().setStatusInfo('lilive bookmarks', '' ) }
        )
    }
}
```
